packages/fastify-vite/config.js

```diff
@@ -232,31 +234,41 @@ function resolveClientModule(root) {
   return null
 }
 
-function resolveRoot(path) {
+function resolveRoot(distDir, path) {
   let root = path
   if (root.startsWith('file:')) {
     root = fileURLToPath(root)
   }
+  let dir = root
   if (stat(root).isFile()) {
-    return dirname(root)
+    dir = dirname(root)
   }
-  return root
+  // Check if running under dist folder
+  const split = dir.split(sep)
+  if (split.at(-1) === distDir) {
+    return split.slice(0, -1).join(sep)
+  }
+  return dir
 }
 
 async function resolveViteConfig(root, dev, { spa, distDir } = {}) {
   const command = 'build'
   const mode = dev ? 'development' : 'production'
   if (!dev) {
+    let viteDistDir = distDir
+    if (!isAbsolute(viteDistDir)) {
+      viteDistDir = join(root, viteDistDir)
+    }
     let viteConfigDistFile
     // Check for top-level dist/ folder
-    viteConfigDistFile = resolve(root, distDir, 'config.json')
+    viteConfigDistFile = resolve(viteDistDir, 'config.json')
```

Comment:
I really think we should name the file `vite-config.json` at least, since `config.json` is too generic and could very easily come into conflict with some other file in user-land that tries to do something unrelated. It would also be easier to find+replace.

Reply:
Good idea, I'll rename to `vite.config.json` for symmetry with `vite.config.js`.